pkg/kapp/cmd/app/deploy.go

```diff
@@ -144,6 +144,10 @@ func (o *DeployOptions) Run() error {
 		return err
 	}
 
+	if o.DeployFlags.Logs {
```

Comment:
This flag is by default true as of now so here we will fetch Pods for every deploy operation. Other possible options here - 
1. Parsing yaml and checking whether `kapp.k14s.io/deploy-log` annotation exists in the provided yaml 
2. Start checking earlier whether the PODs created for the given resource have the annotation or not as we check [here](https://github.com/vmware-tanzu/carvel-kapp/blob/bfa32357c416f669ea5bb20c891baf936e7af038/pkg/kapp/cmd/app/deploy.go#L470).